src/Microsoft.ApplicationInsights/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -178,6 +178,18 @@ public ITelemetry DeepClone()
             return new AvailabilityTelemetry(this);
         }
 
+        /// <inheritdoc/>
+        public void Serialize(ISerializationWriter serializationWriter)
```

Comment:
Everything looks better now.... Now talk about usage of it - is it what our partners expect. I wonder if we need to have separate methods for envelope and separate for `data` part of it. This way envelope structure is not forced on consumer and there is a choice to optimize an envelope structure for specific channels. What do you think?